active/PagerControl/PagerControl.md

```diff
@@ -0,0 +1,405 @@
+
+    <Page.Resources>
+        <CollectionViewSource x:Name="imagesCVS" />
+        <DataTemplate x:Key="ImageTemplate" x:DataType="data:CustomImage">
+            <Image Stretch="UniformToFill" Source="{x:Bind ImageLocation}"
+                   Width="190" Height="130" />
+        </DataTemplate>
+    </Page.Resources>
+
+    <Grid>
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="10*"/>
+            <RowDefinition Height="Auto"/>
+        </Grid.RowDefinitions>
+        <TextBlock Grid.Row="0"
+                   Text="Image Gallery - DataGrid"
+                   Margin="10,0,0,0"
+                   Style="{ThemeResource TitleTextBlockStyle}"/>
+        <controls:DataGrid Grid.Row="1" 
+                           ItemsSource="{x:Bind imagesCVS.View, Mode=OneWay}"
+                           AutoGenerateColumns="True"
+                           AutoGeneratingColumn="DataGrid_AutoGeneratingColumn"/>
+        <data:Pager x:Name="MainPager" Grid.Row="2" PageChanged="Pager_PageChanged" PagerDisplayMode="ButtonPanel"/>
+    </Grid>
+</Page>
+```
+
+C#
+```C#
+        private List<CustomImage> imageList = new List<CustomImage>();
+        private int imagesPerPage = 25;
+
+        public DataGridSamplePage()
+        {
+            this.InitializeComponent();
+            imagesCVS.Source = new ObservableCollection<CustomImage>();
+            Loaded += ListViewSamplePage_Loaded;
+        }
+
+        private async void ListViewSamplePage_Loaded(object sender, RoutedEventArgs e)
+        {
+            imageList = await CustomImage.GenerateImages();
+            MainPager.NumberOfPages = (int)Math.Ceiling((double)imageList.Count / imagesPerPage);
+            UpdateCollection(MainPager.SelectedIndex);
+        }
+
+        private void UpdateCollection(int pageIndex = 0)
+        {
+            List<CustomImage> images = GetPageImages(pageIndex);
+            ((ObservableCollection<CustomImage>)imagesCVS.Source)?.Clear();
+            foreach (CustomImage c in images)
+            {
+                ((ObservableCollection<CustomImage>)imagesCVS.Source)?.Add(c);
+            };
+        }
+
+        private List<CustomImage> GetPageImages(int pageIndex)
+        {
+            if ((pageIndex + 1) * imagesPerPage > imageList.Count)
+            {
+                return imageList.GetRange(pageIndex * imagesPerPage, imageList.Count - (pageIndex * imagesPerPage));
+            }
+
+            return imageList.GetRange(pageIndex * imagesPerPage, imagesPerPage);
+        }
+
+        private void Pager_PageChanged(Pager sender, PageChangedEventArgs args)
+        {
+            if (imageList.Count == 0)
+            {
+                return;
+            }
+
+            UpdateCollection(args.NewPageIndex);
+        }
+
+        private void DataGrid_AutoGeneratingColumn(object sender, Microsoft.Toolkit.Uwp.UI.Controls.DataGridAutoGeneratingColumnEventArgs e)
```

Comment:
We could remove this method

Reply:
Thank you ranjesh this is much cleaner! I want to test it again before we fully publish the documentation but I appreciate the help!

Reply:
Simplified further with async loading of just the items for each page.

<Page
    x:Class="DataGridSamplePage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:controls="using:Microsoft.Toolkit.Uwp.UI.Controls"
    xmlns:controls="using:Microsoft.UI.Xaml.Controls"
    Loaded="OnPageLoaded">

    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition Height="*"/>
            <RowDefinition Height="auto"/>
        </Grid.RowDefinitions>
        <controls:DataGrid x:Name="dataGrid" AutoGenerateColumns="True"/>
        <local:Pager x:Name="pager" Grid.Row="1" PageChanged="OnPageChanged" PagerDisplayMode="NumberPanel"/>
    </Grid>
</Page>

 public sealed partial class DataGridSamplePage : Page
    {
        private const int imagesPerPage = 25;

        public DataGridSamplePage()
        {
            this.InitializeComponent();
        }
    
        private async void OnPageLoaded(object sender, RoutedEventArgs e)
        {
            pager.NumberOfPages = await ImageGenerator.GetNumberOfPages(imagesPerPage);
            dataGrid.ItemsSource = await ImageGenerator.GetImagesInPage(0, imagesPerPage);
        }
        private async void OnPageChanged(Pager sender, PageChangedEventArgs args)
        {
            dataGrid.ItemsSource = await ImageGenerator.GetImagesInPage(args.CurrentPage, imagesPerPage);
        }
    }